### PR TITLE
Port WPE functionalities to Graphene

### PR DESCRIPTION
**Pull request by manju956**
- Ported WPE worker functions viz, GenerateNonce, VerifyUIDSignature and WorkOrderProcess
  to python for Graphene worker
- Modified Graphene WPE get_quote function to include UID generated from KME into its report
- Reorganized common functions/files between WPE and Graphene WPE managers
- Moved common attestation changes like connect_to_ias() and generate_avr() to
  base class EnclaveAttributes
- Added new script to parse the toml having GSC image info to extract MRENCLAVE value and
  persist to separate file

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: common/crypto_utils/avalon_crypto_utils/crypto_utility.py [hunk 1]
```diff
@@ -69,6 +70,21 @@ def base64_to_byte_array(b64_str):
 
 
 # -------------------------------------------------------------------------
+
+def generate_random_bytes(size):
+    """
+    Function to generate random initialization vector
```
Comment (rranjan3) [inline common/crypto_utils/avalon_crypto_utils/crypto_utility.py:76]: Not necessarily an iv. Could be generated for some other purpose as well.

File: enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py [hunk 2]
```diff
@@ -14,30 +14,67 @@
...
 import logging
 
+from avalon_enclave_manager.enclave_attributes import EnclaveAttributes
 logger = logging.getLogger(__name__)
 
 # -------------------------------------------------------------------------
 
 
-class GrapheneWPESignupInfo():
+# class GrapheneWPESignupInfo(EnclaveAttributes):
```
Comment (rranjan3) [inline enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py:23]: Discard comment.

File: enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py [hunk 3]
```diff
@@ -14,30 +14,67 @@
...
+class GrapheneWPESignupInfo(EnclaveAttributes):
     """
     Signup Object that will be used by BaseEnclave Manager
     """
 
 # -------------------------------------------------------------------------
-    def __init__(self, worker_signup_json):
+    def __init__(self, config, worker_signup_json):
```
Comment (rranjan3) [inline enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py:30]: Need to figure a way to common out with graphene_enclave_info.py. The codebase is pretty much same.

File: enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_manager.py [hunk 4]
```diff
@@ -242,9 +266,7 @@ def main(args=None):
 
     try:
         EnclaveManager.parse_command_line(config, remainder)
-        logger.info("Initialize GrapheneWPEManager")
```
Comment (rranjan3) [inline enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_manager.py:245]: Why removed the comments? Anything specific.

File: enclave_manager/avalon_enclave_manager/wpe_common/wo_processor_manager_helper.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+from abc import ABC, abstractmethod
+from utility.jrpc_utility import create_error_response
+from error_code.error_status import WorkOrderStatus
+
+logger = logging.getLogger(__name__)
+
+
+class WOProcessorEnclaveManagerHelper(ABC):
+
+    def __init__(self, config):
```
Comment (rranjan3) [inline enclave_manager/avalon_enclave_manager/wpe_common/wo_processor_manager_helper.py:29]: Why pass the config if not being used?

File: enclave_manager/avalon_enclave_manager/wpe_common/wpe_requester.py [hunk 6]
```diff
@@ -172,6 +168,7 @@ def preprocess_work_order(self, wo_request, encryption_key):
             @returns result - Result from KME that includes the workorder
                               key info. error response, in case of failure.
         """
+        logger.info("inside WPE REQ:: preprocess")
```
Comment (rranjan3) [inline enclave_manager/avalon_enclave_manager/wpe_common/wpe_requester.py:171]: Probably debug log.

File: tc/graphene/python_worker/avalon_worker/base_work_order_processor.py [hunk 7]
```diff
@@ -275,31 +292,23 @@ def _get_mrenclave(self):
             mrenclave value of worker enclave as hex string.
             If Intel SGX environment is not present returns empty string.
         """
+        logger.info("GET mr enclave")
         mrenclave = self.sgx_attestation.get_mrenclave()
+        logger.info(
```
Comment (rranjan3) [inline tc/graphene/python_worker/avalon_worker/base_work_order_processor.py:297]: Debug log!

**Review by rranjan3 — CHANGES_REQUESTED** [review]

File: common/crypto_utils/avalon_crypto_utils/crypto_utility.py (hunk 1 above)
Comment (manju956) [reply common/crypto_utils/avalon_crypto_utils/crypto_utility.py]: Agree, rephrased function comment

File: enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py (hunk 2 above)
Comment (manju956) [reply enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py]: Done

File: enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py (hunk 3 above)
Comment (manju956) [reply enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_enclave_info.py:29]: Yes, refactoring needs to be looked at. Could be addressed as a separate PR.

File: enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_manager.py (hunk 4 above)
Comment (manju956) [reply enclave_manager/avalon_enclave_manager/graphene_wpe/graphene_wpe_manager.py:245]: It was accidental. Reverted the log.

File: enclave_manager/avalon_enclave_manager/wpe_common/wo_processor_manager_helper.py (hunk 5 above)
Comment (manju956) [reply enclave_manager/avalon_enclave_manager/wpe_common/wo_processor_manager_helper.py]: Removed.

File: enclave_manager/avalon_enclave_manager/wpe_common/wpe_requester.py (hunk 6 above)
Comment (manju956) [reply enclave_manager/avalon_enclave_manager/wpe_common/wpe_requester.py]: Added while debugging functionality. Not really useful, removed it

File: tc/graphene/python_worker/avalon_worker/base_work_order_processor.py (hunk 7 above)
Comment (manju956) [reply tc/graphene/python_worker/avalon_worker/base_work_order_processor.py]: Added while debugging functionality. Not really useful, removed it

**Review by rranjan3 — APPROVED** [review]

**Pull request merged**